lib/Compiler.js

```diff
@@ -213,10 +212,14 @@ class Compiler extends Tapable {
 		const startTime = Date.now();
 
 		this.running = true;
-
+		const onCompilationErrors = compilation => {
+			if (compilation.errors.length && compilation.bail && !this.watchMode) {
+				finalCallback(...compilation.errors);
```

Comment:
You can't spread `errors` here:
- 1st argument must be an error ✅ 
- 2nd argument must be a `Stat` instance ❌

The above contract will fail if `errors` contains 2 or more errors. Instead you should report the first compilation error.

```suggestion
				finalCallback(compilation.errors[0]);
```

Reply:
Will do.